### PR TITLE
Verify product dependencies w/ target platform, vs. hardcoded `macOS`

### PR DESCRIPTION
**Pull request by peterkos**
Product dependencies are now verified using the `buildEnvironment`'s platform, instead of the hardcoded `macOS` platform.

This builds on the work of #6732, now allowing cross-compilation of product dependencies 🎉

### Motivation:
Let's say we're cross-compiling macOS -> iOS simulator, with a host system of `macOS v13`.
```swift
let package = Package(
    name: "SwiftFlashlight",
    products: [.library(...)],
    platforms: [.iOS(.v16)],
    dependencies: [
        // internally, for iOS(.v16), macOS(.v14)
        .package(url: "probably/alamofire", exact: "1.2.3")
    ],
    // ...
)
```

As long as we set our triple & SDK, we _should_ be good... (and, with #6828, only the triple needed)
```fish
> swift build --triple arm64-apple-ios-simulator \
              --sdk "$(xcrun --sdk iphonesimulator --show-sdk-path)"
...
error: the library 'SwiftFlashlight' requires macos 10.13,
but depends on the product 'probably/alamofire' which requires macos 10.14;  [...]
```

Even though `SwiftFlashlight` only supports `iOS(.v16)`, [SupportedPlatform](https://github.com/apple/swift-package-manager/blob/main/Documentation/PackageDescription.md#supportedplatform) "[by] default, [...] assigns a predefined minimum deployment version for each supported platforms unless you configure supported platforms [...]".  As a side-effect, `macOS` is inferred as a valid platform. (This is why no "target/platform mismatch" error was thrown.)

Previously, product dependencies were verified against `macOS`, rather than the target platform. Since `macOS` is always inferred, this surfaced as a "woah! your dependencies are out of sync".

```swift

let swiftFlashlight = Package(
    platforms: [
        .iOS(.v16),
        .macOS(.v13) // <-- inferred-|
    ],                               |
)                                    |-- // since --triple is .iOS,
                                     |   // shouldn't be comparing
let probably/alamofire = Package (   |   // macOS.
    platforms: [                     |
        iOS(.v16),                   |
        macOS(.v14)] // <-- inferred-|
    ]
)


```

Now, the intention of `--triple` is respected 🐱

### Modifications:
- `validateDeploymentVersionOfProductDependency` accepts a `buildEnvironment` containing the target platform

### Result:

Product dependencies can now be compiled for `darwin` other than macOS -- i.e, iOS :)

### Future work
- Target-level or product-level platform support is still up in the air
- This only applies to `triple.isDarwin()` -- perhaps a next step of, verifying dependencies against non-darwin?

### CODE REVIEW

**Comment by peterkos** [comment]
This was a fun one to chase down 😅 
(note to self: `Apple Swift version 5.9 (swift-5.9-RELEASE)`)

Some questions:
- I tried running `swiftformat` but it changed 600+ files -- just `BuildPlan` was ~300 lines. Thoughts?
- Happy to add some tests around this; wanted to get some thoughts before diving in there :)

**Comment by MaxDesiatov** [comment]
> I tried running swiftformat but it changed 600+ files -- just BuildPlan was ~300 lines 😵, thoughts?

Please run `Utilities/soundness.sh` script, which will only format files that were changed.

> Happy to add some tests around this; wanted to get some thoughts before diving in there :)

Yes, please add the tests, that would be great. Thanks!

**Comment by peterkos** [comment]
Thanks! I'll take a crack at it soon :)

**Comment by peterkos** [comment]
@MaxDesiatov Apologies for the delay on this! One test added, one test modified 🧪 

Since `utilities/soundness.sh` works from the `git diff`, I added a newline in each file I modified + ran the script as a separate commit for readability. As you can see, those files haven't had a run in a _while_, heh.

**Comment by MaxDesiatov** [comment]
Thanks! Just as heads up, this will conflict with https://github.com/apple/swift-package-manager/pull/7161

**Comment by peterkos** [comment]
Resolved conflicts 👍 

Looks like `soundness.sh` disagrees with how Swift requires raw string literals to be formatted tho. One example:
```swift
// BuildPlanTests L#5749
func testArtifactsArchiveBinaryTargets(...) {
    // ...
                        "path": "all-platforms/mytool",
                        "supportedTriples": ["\(
                            artifactTriples.map(\.tripleString)
                                .joined(separator: "\", \"")
    )"]
                }
            ]
        }
    }
}
                    """
                //   ^ swiftformat throws this here,
                //     but is a swift build error :(
```

I've manually reverted the script's raw string changes for now, but it seems like the way swiftformat expands inline closures (among other things) could be improved -- that's my guess 🚀 

Also the `soundness.sh` script didn't print any output unless I tweaked L60. Assuming it's a me-issue, but can push up that change if useful.
```sh
swiftformat $changed_file > dev/null 2>&1 # old
swiftformat $changed_file 2>&1            # new
```

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

File: Sources/SPMTestSupport/MockBuildTestHelper.swift [hunk 1]
```diff
@@ -134,23 +133,29 @@ enum BuildError: Swift.Error {
...
-        self.productMap = try Dictionary(throwingUniqueKeysWithValues: plan.buildProducts.compactMap { $0 as? Build.ProductBuildDescription }.map{ ($0.product.name, $0) })
-        self.targetMap = try Dictionary(throwingUniqueKeysWithValues: plan.targetMap.map{ ($0.0.name, $0.1) })
+        self
+            .productMap = try Dictionary(
+                throwingUniqueKeysWithValues: plan.buildProducts
+                    .compactMap { $0 as? Build.ProductBuildDescription }.map { (
+                        $0.product.name,
+                        $0
+                    ) }
+            )
```
Comment (MaxDesiatov) [inline Sources/SPMTestSupport/MockBuildTestHelper.swift:149]:
```suggestion
        self.productMap = try Dictionary(
                throwingUniqueKeysWithValues: plan.buildProducts
                    .compactMap { $0 as? Build.ProductBuildDescription }
                    .map { ($0.product.name, $0) }
        )
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 2]
```diff
@@ -1057,9 +1143,22 @@ final class BuildPlanTests: XCTestCase {
         let buildPath = result.plan.destinationBuildParameters.dataPath.appending(components: "release")
 
         let exe = try result.target(for: "exe").swiftTarget().compileArguments()
-        XCTAssertMatch(exe, ["-swift-version", "4", "-O", .equal(j), "-DSWIFT_PACKAGE", "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))", .anySequence, "-g"])
+        XCTAssertMatch(
+            exe,
+            [
+                "-swift-version",
+                "4",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:1150]:
```suggestion
                "-swift-version", "4",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 3]
```diff
@@ -1057,9 +1143,22 @@ final class BuildPlanTests: XCTestCase {
...
+                "-O",
+                .equal(self.j),
+                "-DSWIFT_PACKAGE",
+                "-module-cache-path",
+                "\(buildPath.appending(components: "ModuleCache"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:1155]:
```suggestion
                "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 4]
```diff
@@ -1134,9 +1235,22 @@ final class BuildPlanTests: XCTestCase {
         let buildPath = result.plan.destinationBuildParameters.dataPath.appending(components: "release")
 
         let exe = try result.target(for: "exe").swiftTarget().compileArguments()
-        XCTAssertMatch(exe, ["-swift-version", "4", "-O", .equal(j), "-DSWIFT_PACKAGE", "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))", .anySequence, "-g"])
+        XCTAssertMatch(
+            exe,
+            [
+                "-swift-version",
+                "4",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:1242]:
```suggestion
                "-swift-version", "4",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 5]
```diff
@@ -1134,9 +1235,22 @@ final class BuildPlanTests: XCTestCase {
...
+                "-O",
+                .equal(self.j),
+                "-DSWIFT_PACKAGE",
+                "-module-cache-path",
+                "\(buildPath.appending(components: "ModuleCache"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:1247]:
```suggestion
                "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 6]
```diff
@@ -1831,39 +1980,46 @@ final class BuildPlanTests: XCTestCase {
...
+                "-o",
+                buildPath.appending(components: "PkgPackageTests.xctest", "Contents", "MacOS", "PkgPackageTests")
+                    .pathString,
+                "-module-name", "PkgPackageTests",
+                "-Xlinker", "-bundle",
+                "-Xlinker", "-rpath", "-Xlinker", "@loader_path/../../../",
+                "@\(buildPath.appending(components: "PkgPackageTests.product", "Objects.LinkFileList"))",
+            ] +
+                rpathsForBackdeployment +
+                ["-target", "\(hostTriple.tripleString(forPlatformVersion: version))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2015]:
```suggestion
            ] + rpathsForBackdeployment + [
                "-target", "\(hostTriple.tripleString(forPlatformVersion: version))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 7]
```diff
@@ -1831,39 +1980,46 @@ final class BuildPlanTests: XCTestCase {
...
+                 "-Xlinker", "-add_ast_path", "-Xlinker",
+                 buildPath.appending(components: "Modules", "Foo.swiftmodule").pathString,
+                 "-Xlinker", "-add_ast_path", "-Xlinker",
+                 buildPath.appending(components: "Modules", "FooTests.swiftmodule").pathString,
+                 "-g"]
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2020]:
```suggestion
                 buildPath.appending(components: "Modules", "FooTests.swiftmodule").pathString,
                 "-g"
             ]
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 8]
```diff
@@ -1926,9 +2084,24 @@ final class BuildPlanTests: XCTestCase {
 
         let exe = try result.target(for: "exe").swiftTarget().compileArguments()
 
-        XCTAssertMatch(exe, [.anySequence, "-swift-version", "4", "-O", .equal(j), "-DSWIFT_PACKAGE", "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))", .anySequence, "-g", .anySequence])
+        XCTAssertMatch(
+            exe,
+            [
+                .anySequence,
+                "-swift-version",
+                "4",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2092]:
```suggestion
                "-swift-version", "4",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 9]
```diff
@@ -1926,9 +2084,24 @@ final class BuildPlanTests: XCTestCase {
...
+                "-O",
+                .equal(self.j),
+                "-DSWIFT_PACKAGE",
+                "-module-cache-path",
+                "\(buildPath.appending(components: "ModuleCache"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2097]:
```suggestion
                "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 10]
```diff
@@ -2267,9 +2447,28 @@ final class BuildPlanTests: XCTestCase {
 
         let buildPath = result.plan.productsBuildPath
 
-        XCTAssertMatch(try result.target(for: "exe").swiftTarget().compileArguments(), ["-swift-version", "4", "-enable-batch-mode", "-Onone", "-enable-testing", .equal(j), "-DSWIFT_PACKAGE", "-DDEBUG", "-Xcc", "-fmodule-map-file=\(Clibgit.appending(components: "module.modulemap"))", "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))", .anySequence, "-g", .anySequence])
+        XCTAssertMatch(
+            try result.target(for: "exe").swiftTarget().compileArguments(),
+            [
+                "-swift-version",
+                "4",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2454]:
```suggestion
                "-swift-version", "4",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 11]
```diff
@@ -2267,9 +2447,28 @@ final class BuildPlanTests: XCTestCase {
...
+                "-enable-batch-mode",
+                "-Onone",
+                "-enable-testing",
+                .equal(self.j),
+                "-DSWIFT_PACKAGE",
+                "-DDEBUG",
+                "-Xcc",
+                "-fmodule-map-file=\(Clibgit.appending(components: "module.modulemap"))",
+                "-module-cache-path",
+                "\(buildPath.appending(components: "ModuleCache"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2464]:
```suggestion
                "-Xcc", "-fmodule-map-file=\(Clibgit.appending(components: "module.modulemap"))",
                "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 12]
```diff
@@ -2280,10 +2479,11 @@ final class BuildPlanTests: XCTestCase {
             "@\(buildPath.appending(components: "exe.product", "Objects.LinkFileList"))",
             "-Xlinker", "-rpath", "-Xlinker", "/fake/path/lib/swift-5.5/macosx",
             "-target", defaultTargetTriple,
-            "-Xlinker", "-add_ast_path", "-Xlinker", buildPath.appending(components: "exe.build", "exe.swiftmodule").pathString,
+            "-Xlinker", "-add_ast_path", "-Xlinker",
+            buildPath.appending(components: "exe.build", "exe.swiftmodule").pathString,
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2483]:
```suggestion
            "-Xlinker", "-add_ast_path",
            "-Xlinker", buildPath.appending(components: "exe.build", "exe.swiftmodule").pathString,
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 13]
```diff
@@ -2421,7 +2626,8 @@ final class BuildPlanTests: XCTestCase {
             "@\(buildPath.appending(components: "Foo.product", "Objects.LinkFileList"))",
             "-Xlinker", "-rpath", "-Xlinker", "/fake/path/lib/swift-5.5/macosx",
             "-target", defaultTargetTriple,
-            "-Xlinker", "-add_ast_path", "-Xlinker", buildPath.appending(components: "Foo.build", "Foo.swiftmodule").pathString,
+            "-Xlinker", "-add_ast_path", "-Xlinker",
+            buildPath.appending(components: "Foo.build", "Foo.swiftmodule").pathString,
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2630]:
```suggestion
            "-Xlinker", "-add_ast_path",
            "-Xlinker", buildPath.appending(components: "Foo.build", "Foo.swiftmodule").pathString,
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 14]
```diff
@@ -2436,10 +2642,11 @@ final class BuildPlanTests: XCTestCase {
             "@\(buildPath.appending(components: "Bar-Baz.product", "Objects.LinkFileList"))",
             "-Xlinker", "-rpath", "-Xlinker", "/fake/path/lib/swift-5.5/macosx",
             "-target", defaultTargetTriple,
-            "-Xlinker", "-add_ast_path", "-Xlinker", buildPath.appending(components: "Modules", "Bar.swiftmodule").pathString,
+            "-Xlinker", "-add_ast_path", "-Xlinker",
+            buildPath.appending(components: "Modules", "Bar.swiftmodule").pathString,
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2646]:
```suggestion
            "-Xlinker", "-add_ast_path",
            "-Xlinker", buildPath.appending(components: "Modules", "Bar.swiftmodule").pathString,
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 15]
```diff
@@ -2538,49 +2747,84 @@ final class BuildPlanTests: XCTestCase {
         let buildPath = result.plan.productsBuildPath
 
         let exe = try result.target(for: "exe").swiftTarget().compileArguments()
-        XCTAssertMatch(exe, ["-swift-version", "4", "-enable-batch-mode", "-Onone", "-enable-testing", .equal(j), "-DSWIFT_PACKAGE", "-DDEBUG", "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))", .anySequence, "-g", .anySequence])
+        XCTAssertMatch(
+            exe,
+            [
+                "-swift-version",
+                "4",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2754]:
```suggestion
                "-swift-version", "4",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 16]
```diff
@@ -2538,49 +2747,84 @@ final class BuildPlanTests: XCTestCase {
...
+                "-enable-batch-mode",
+                "-Onone",
+                "-enable-testing",
+                .equal(self.j),
+                "-DSWIFT_PACKAGE",
+                "-DDEBUG",
+                "-module-cache-path",
+                "\(buildPath.appending(components: "ModuleCache"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2762]:
```suggestion
                "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 17]
```diff
@@ -2538,49 +2747,84 @@ final class BuildPlanTests: XCTestCase {
...
-                "-Xlinker", "-rpath", "-Xlinker", "@loader_path",
-                "@\(buildPath.appending(components: "lib.product", "Objects.LinkFileList"))",
-                "-Xlinker", "-rpath", "-Xlinker", "/fake/path/lib/swift-5.5/macosx",
-                "-target", defaultTargetTriple,
-                "-Xlinker", "-add_ast_path", "-Xlinker", buildPath.appending(components: "Modules", "lib.swiftmodule").pathString,
+        XCTAssertMatch(
+            lib,
+            [
+                "-swift-version",
+                "4",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2774]:
```suggestion
                "-swift-version", "4",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 18]
```diff
@@ -2538,49 +2747,84 @@ final class BuildPlanTests: XCTestCase {
...
+                "-enable-batch-mode",
+                "-Onone",
+                "-enable-testing",
+                .equal(self.j),
+                "-DSWIFT_PACKAGE",
+                "-DDEBUG",
+                "-module-cache-path",
+                "\(buildPath.appending(components: "ModuleCache"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2782]:
```suggestion
                "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 19]
```diff
@@ -2538,49 +2747,84 @@ final class BuildPlanTests: XCTestCase {
...
-            ]
+        let linkArguments = [
+            result.plan.destinationBuildParameters.toolchain.swiftCompilerPath.pathString,
+            "-L", buildPath.pathString,
+            "-o", buildPath.appending(components: "lib.dll").pathString,
+            "-module-name", "lib",
+            "-emit-library",
+            "@\(buildPath.appending(components: "lib.product", "Objects.LinkFileList"))",
+            "-target", defaultTargetTriple,
+            "-g", "-use-ld=lld", "-Xlinker", "-debug:dwarf",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:2814]:
```suggestion
            "-g", "-use-ld=lld",
            "-Xlinker", "-debug:dwarf",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 20]
```diff
@@ -3116,8 +3374,9 @@ final class BuildPlanTests: XCTestCase {
         let lib = try result.target(for: "lib").clangTarget()
         let args = [
             "-target", "x86_64-unknown-windows-msvc", "-O0",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:3376]:
```suggestion
            "-target", "x86_64-unknown-windows-msvc",
            "-O0",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 21]
```diff
@@ -3116,8 +3374,9 @@ final class BuildPlanTests: XCTestCase {
...
-            "-DSWIFT_PACKAGE=1", "-DDEBUG=1", "-fblocks", "-I", Pkg.appending(components: "Sources", "lib", "include").pathString,
-            "-gdwarf"
+            "-DSWIFT_PACKAGE=1", "-DDEBUG=1", "-fblocks", "-I",
+            Pkg.appending(components: "Sources", "lib", "include").pathString,
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:3378]:
```suggestion
            "-DSWIFT_PACKAGE=1",
            "-DDEBUG=1",
            "-fblocks",
            "-I", Pkg.appending(components: "Sources", "lib", "include").pathString,
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 22]
```diff
@@ -3129,14 +3388,14 @@ final class BuildPlanTests: XCTestCase {
...
             "-Onone",
             "-enable-testing",
-            .equal(j),
+            .equal(self.j),
             "-DSWIFT_PACKAGE", "-DDEBUG",
             "-Xcc", "-fmodule-map-file=\(buildPath.appending(components: "lib.build", "module.modulemap"))",
             "-Xcc", "-I", "-Xcc", "\(Pkg.appending(components: "Sources", "lib", "include"))",
             "-module-cache-path", "\(buildPath.appending(components: "ModuleCache"))",
             .anySequence,
             "-g", "-use-ld=lld", "-Xcc", "-gdwarf",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:3397]:
```suggestion
            "-g",
            "-use-ld=lld",
            "-Xcc", "-gdwarf",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 23]
```diff
@@ -3145,8 +3404,8 @@ final class BuildPlanTests: XCTestCase {
             "-o", buildPath.appending(components: "exe.exe").pathString,
             "-module-name", "exe", "-emit-executable",
             "@\(buildPath.appending(components: "exe.product", "Objects.LinkFileList"))",
-             "-target", "x86_64-unknown-windows-msvc",
-             "-g", "-use-ld=lld", "-Xlinker", "-debug:dwarf",
+            "-target", "x86_64-unknown-windows-msvc",
+            "-g", "-use-ld=lld", "-Xlinker", "-debug:dwarf",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:3408]:
```suggestion
            "-g",
            "-use-ld=lld",
            "-Xlinker", "-debug:dwarf",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 24]
```diff
@@ -4373,16 +4802,40 @@ final class BuildPlanTests: XCTestCase {
...
         let fooTarget = try result.target(for: "Foo").swiftTarget().compileArguments()
         #if os(macOS)
-          XCTAssertMatch(fooTarget, [.anySequence, "-emit-objc-header", "-emit-objc-header-path", "/path/to/build/debug/Foo.build/Foo-Swift.h", .anySequence])
+        XCTAssertMatch(
+            fooTarget,
+            [
+                .anySequence,
+                "-emit-objc-header",
+                "-emit-objc-header-path",
+                "/path/to/build/debug/Foo.build/Foo-Swift.h",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:4811]:
```suggestion
                "-emit-objc-header-path", "/path/to/build/debug/Foo.build/Foo-Swift.h",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 25]
```diff
@@ -4373,16 +4802,40 @@ final class BuildPlanTests: XCTestCase {
...
+        )
         #else
-          XCTAssertNoMatch(fooTarget, [.anySequence, "-emit-objc-header", "-emit-objc-header-path", "/path/to/build/debug/Foo.build/Foo-Swift.h", .anySequence])
+        XCTAssertNoMatch(
+            fooTarget,
+            [
+                .anySequence,
+                "-emit-objc-header",
+                "-emit-objc-header-path",
+                "/path/to/build/debug/Foo.build/Foo-Swift.h",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:4822]:
```suggestion
                "-emit-objc-header-path", "/path/to/build/debug/Foo.build/Foo-Swift.h",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 26]
```diff
@@ -5134,7 +5622,15 @@ final class BuildPlanTests: XCTestCase {
 
         let clibraryBasicArguments = try result.target(for: "CLibrary").clangTarget().basicArguments(isCXX: false)
         XCTAssertMatch(clibraryBasicArguments, [.anySequence, "-F", "\(buildPath)", .anySequence])
-        XCTAssertMatch(clibraryBasicArguments, [.anySequence, "-I", "\(Pkg.appending(components: "StaticLibrary.xcframework", "\(platform)-\(arch)", "Headers"))", .anySequence])
+        XCTAssertMatch(
+            clibraryBasicArguments,
+            [
+                .anySequence,
+                "-I",
+                "\(Pkg.appending(components: "StaticLibrary.xcframework", "\(platform)-\(arch)", "Headers"))",
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:5630]:
```suggestion
                "-I", "\(Pkg.appending(components: "StaticLibrary.xcframework", "\(platform)-\(arch)", "Headers"))",
```

File: Tests/BuildTests/BuildPlanTests.swift [hunk 27]
```diff
@@ -5520,11 +6038,13 @@ final class BuildPlanTests: XCTestCase {
             "@\(buildPath.appending(components: "exe.product", "Objects.LinkFileList"))",
             "-Xlinker", "-rpath", "-Xlinker", "/fake/path/lib/swift-5.5/macosx",
             "-target", defaultTargetTriple,
-            "-Xlinker", "-add_ast_path", "-Xlinker", buildPath.appending(components: "Modules", "lib.swiftmodule").pathString,
-            "-Xlinker", "-add_ast_path", "-Xlinker", buildPath.appending(components: "exe.build", "exe.swiftmodule").pathString,
+            "-Xlinker", "-add_ast_path", "-Xlinker",
+            buildPath.appending(components: "Modules", "lib.swiftmodule").pathString,
+            "-Xlinker", "-add_ast_path", "-Xlinker",
+            buildPath.appending(components: "exe.build", "exe.swiftmodule").pathString,
```
Comment (MaxDesiatov) [inline Tests/BuildTests/BuildPlanTests.swift:6044]:
```suggestion
            "-Xlinker", "-add_ast_path",
            "-Xlinker", buildPath.appending(components: "Modules", "lib.swiftmodule").pathString,
            "-Xlinker", "-add_ast_path",
            "-Xlinker", buildPath.appending(components: "exe.build", "exe.swiftmodule").pathString,
```

**Comment by MaxDesiatov** [comment]
@swift-ci test

**Review by MaxDesiatov — APPROVED** [review]
Thank you!

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Pull request merged**